code/modules/mob/living/simple_animal/friendly/mouse.dm

```diff
@@ -44,6 +44,8 @@
 		..(1)
 		var/obj/item/trash/deadmouse/M = new(src.loc)
 		M.icon_state = icon_dead
+		if(name != initial(name))
```

Comment:
I don't think this fixes it. Map edited mice will have their initial() name different than the coded name, so those mice, such as Tom, will generate a 'mouse' on death. Just ditch the if check and always set the 'M.name'.